telemetry/packages/public-app/components/timer.tsx

```diff
@@ -0,0 +1,68 @@
+import { Card, Metric, Text } from '@tremor/react';
+
+import { useEffect, useState } from 'react';
+import { Badge, BadgeDelta, Callout } from '@tremor/react';
+import { useQuery } from 'react-query';
+
+export function DigitalTimer(): JSX.Element {
```

Comment:
```suggestion
export function DigitalTimer() {
```
again no explicit return type needed - TypeScript will infer this for us

Reply:
not knowing much about typescript, is this actually good practice? surely you want return types explicit so you get type errors in the right places should you change the internals of the function.
or is TS smart enough to always figure out that the return value is the wrong type instead of e.g. where that value gets passed to another function?

Reply:
General consensus is that it's just not necessary almost all of the time - and I'd argue particularly with React components. React nodes can be null, strings, false as well as JSX so it's hard to write a component with an invalid return type for a start. If you do it'll give you an error where you're using it anyway (which isn't that inconvenient vs. the clutter of having the forced return type). Good Matt Pocock video on it: [https://youtu.be/nwSe95uFN8E?si=eDiirjhUPvAhUlok](https://youtu.be/nwSe95uFN8E?si=eDiirjhUPvAhUlok)

Reply:
if that's the standard then go ahead, and fair point on nodes being weird, but I very strongly disagree with that view of the purpose of a function signature

we can argue about it at the hackathon lol